Add CITEseq postprocessing to QC report

sjspielman opened the pull request:
Closes #336 

This PR updates the QC report to have a bit more information about CITEseq post-processing. There wasn't much that really needed to be done it seemed in the end? I've done where what I think is the minimum we need (note one `TODO` comment in the `qc_report.rmd` file which is a question for reviewers!), but we could add more details, for example, maybe the number of cells removed based on RNA info vs. ADT info. Any recommendations for other stats to include would be helpful! Tagging a couple people for review to crowd source any other ideas for info to include.

While I was here, I also fixed a couple things..
- The `scpca_filter` column is present in both `_filtered` and `_processed` RDS files, so we remove a bit of wrangling
- I cleaned up a couple spacing around parentheses for styling 
- I tweaked, in line 444 of `qc_report.rmd`, a phrase that I think should not be there - I deleted the bit "calculated by `miQC`", since we then proceed to explain that maybe it wasn't calculated by miQC! Was this a good move on my part?

When testing this, I found that running `nextflow` with `-resume` did not want to use the updated rmds (is the `template` directory not copied over fresh, I wonder? any insights as to why `nextflow` didn't want to use? i wasn't pulling anything fresh from git, just running `main.nf`...). So, I just locally knitted with whatever package versions I have, which is maybe why we have the warning on top, unless we do need to update the `mutate()` code there. Local knitting is definitely responsible for the title though! Here's the report now:  
[qc_report.html.zip](https://github.com/AlexsLemonade/scpca-nf/files/11618036/qc_report.html.zip)

jashapiro commented on templates/qc_report/qc_report.rmd:
````diff
@@ -426,26 +441,31 @@ In such situations, the calculated probability of compromise may not be valid (s
 
+```{r eval = has_cite, results='asis'}
+glue::glue(
+"Next, additional low quality cells were removed based on ADT counts (<code>cleanTagCounts</code>), such that cells with high levels of ambient contamination and/or isotype control tags (if present) were removed."
````
I think you can still use markdown here. Eitehr way, in keeping with previous usage, we should probably make this a full reference with a link to the package.

I'm also splitting into two sentences here. 

Update: I just realized that you were probably referring to the text label above with your code! That is fine, but now that I think about it, we should probably make that more descriptive, so that the `scpca_filter_method` encodes whether negative controls were used or not. 

```suggestion
"Next, additional low quality cells were removed based on ADT counts ([`DropletUtils::cleanTagCounts()`](https://bioconductor.org/packages/release/bioc/html/DropletUtils.html)).
Cells with high levels of ambient contamination and/or isotype control tags (if present) were removed."
```

jashapiro reviewed (commented):
Looks good overall, though I realized mid-review that we may want to be more specific in the filtering label we apply in the workflow. I think it is worth surfacing up front if the filtering used negative controls or not.

allyhawkins commented:
So I took a look at this and I have some thoughts which may be kind of crazy, so you can tell me if they are not feasible at this time. You all may disagree with me but... Based on my understanding of the changes that have taken place, for any library that contains CITE-seq data we have now filtered on both the RNA and the ADT counts so that the final `processed.rds` file contains only cells that pass both of those thresholds? So that means only those cells are included in normalization and PCA/UMAP? 

If the above is true and we don't want to change that then the changes I'm about to propose will not be as applicable... 
However, I can very much argue for having `processed.rds` files that contain the cells that have been filtered just on RNA. The CITE-seq library is handled separately from the gene expression library, so there could be various reasons why you throw out a cell that does poorly in CITE-seq but want to keep that cell to look at it's gene expression. So if users want to look at these samples regardless of CITE-seq, then they are going to lose all of those cells that have good quality RNA data. I noticed this because of the miQC plot where there are a lot of cells getting labeled as remove that I would want to keep. 

Because of that, I think it's important to separate the RNA and the ADT filtering here and actually move all of the mention of ADT filtering and downstream plots that would incorporate ADT filtering to the CITE-seq specific section. Then you show the data as a result of just analyzing the RNA data vs. RNA plus ADT analysis. Again, this may be an unpopular opinion and I know it's later in the game, but I am worried about losing information for RNA that we want to keep.

sjspielman commented on templates/qc_report/qc_report.rmd:
```diff
@@ -227,18 +227,33 @@ basic_statistics <- tibble::tibble(
+if (has_processed) {
   processed_meta <- metadata(processed_sce)
   
+  # TODO: Do we want to separate RNA and ADT rows? ADT is always the same, so may not be needed, but 
```
The reason I split it is really only because I worried it would look confusing, especially since the header for this section is RNA-sequencing. Of course, RNA is also being filtered, just that ADT information is being used to determine the filtering... @allyhawkins, any stylistic opinions here? 

I definitely take your point about future breakage, though! I had thought at one point about whether we might actually want two different metadata fields for this (such as, `scpca_filter_method_rna` and `scpca_filter_method_adt`), but that will already break enough code to not be worth it! Plus, while we need to keep track of which RNA filtering method was used, we only have 1 ADT filtering method so there's less ambiguity.

sjspielman commented:
@allyhawkins I don't think this is "crazy" at all! You made a crux point here - 

> So if users want to look at these samples regardless of CITE-seq, then they are going to lose all of those cells that have good quality RNA data. I noticed this because of the miQC plot where there are a lot of cells getting labeled as remove that I would want to keep.

My immediate question now is whether this implies we need a 4th SCE export. This may not be necessary since there's a `discard` column in the CITE-seq colData that can be used to filter more cells out, but users would have to dig in themselves.

jashapiro commented:
> @allyhawkins I don't think this is "crazy" at all! You made a crux point here -
> 
> > So if users want to look at these samples regardless of CITE-seq, then they are going to lose all of those cells that have good quality RNA data. I noticed this because of the miQC plot where there are a lot of cells getting labeled as remove that I would want to keep.
> 
> My immediate question now is whether this implies we need a 4th SCE export. This may not be necessary since there's a `discard` column in the CITE-seq colData that can be used to filter more cells out, but users would have to dig in themselves.

Memories... I feel like this was a discussion a while ago when we initially decided not to worry about ADT filtering at that time! But I agree 100% that we would not want to make this a third export.

I think one option is that we modify the `scpca_filter` to be more fine grained than `keep` and `remove` to reflect which components are considered "okay". The idea would be that we could add a `remove_adt` or something like that which could be easily filtered on without diving into the `altExp`, but would be kept in the processed data. We could then add a second plot (or update this one) to show which cells would be filtered by ADT data specifically.

allyhawkins commented:
> My immediate question now is whether this implies we need a 4th SCE export. This may not be necessary since there's a discard column in the CITE-seq colData that can be used to filter more cells out, but users would have to dig in themselves.

I think my ideal scenario would be having a separate file that contains the ADT processed data that has been filtered and normalized using ADT counts rather than RNA. However, that means a lot of changes including on the actual portal end and I don't think we want to take that route, at least not right now. 

In the mean time, I think your suggestion where we only filter on RNA counts and then include a separate column indicating if cells would be filtered using ADT counts would be sufficient. We would then need to add that to user facing documentation. We did a similar thing before the processed data even existed where we had a `ccdl_filter` column present in the `filtered.rds` files to suggest which cells to remove. So here we would do the same with the CITE-seq. This would allow us to make a separate plot showing filtering of both RNA + ADT in the CITE-seq section without actually removing any cells that are informative when it comes to RNA.

jashapiro commented:
One other thought: we probably want UMAPs with the ADT expression! Maybe pick the 4 with the greatest variation and plot those?

sjspielman commented:
#moreUMAPSmorebetter

sjspielman commented on templates/qc_report/qc_report.rmd:
```diff
@@ -125,52 +125,53 @@ if(has_multiplex){
-  adt_exp <- altExp(filtered_sce, "adt")
+
+if (has_adt) {
+  adt_exp <- altExp(processed_sce, "adt")
```
noting this is needed for UMAPs, and it has no other effect on output.

sjspielman replied:
Noting this was not, indeed, true since `has_processed` could be false! It was fixed in recent commits.

sjspielman commented:
And we're back for more QC report updates! I've made the following changes:

- In `post_process_sce.R`, I now also calculate the most variable ADTs. I'm not sure about a modality-specific approach here, so I just did the same approach as for genes. I also stashed this in metadata as `highly_variable_adts` for later use in the QC report. 
  - ⚠️ If we keep this strategy, we'll need to add this metadata variable to documentation as well.
- In the QC report itself, I fixed a couple minor spacing spots and changed `~` -> `vars()` for faceting, since I was there.
- In the QC report, I also added a section in the `cite_qc.rmd` for a UMAP of top 4 variable ADTs, using the vector that is now in the metadata, and I tweaked some plot size aspects for visibility.
  -  I left one `TODO` here - currently all cells are being plotted, including those with `NA` logcounts. We could filter those cells out before plotting, but the points are so small that you don't really perceive the `NA` points anyways? 

HTML - [qc_report.html.zip](https://github.com/AlexsLemonade/scpca-nf/files/11708350/qc_report.html.zip)

allyhawkins commented on templates/qc_report/cite_qc.rmd:
```diff
@@ -35,19 +35,114 @@ knitr::kable(adt_information, align = 'r') |>
+
+basic_statistics <- tibble::tibble(
+    "Method used to identify cells targeted for filtering" = format(processed_meta$adt_scpca_filter_method),
+    "Number of cells targeted for filtering"  = format(filtered_cell_count),
```
```suggestion
    "Number of cells that pass filtering thresholds"  = format(filtered_cell_count),
```
I think we probably want to be consistent with how we report in the RNA section, which is the cells that are present after filtering. Also the targeted wording was confusing to me.

sjspielman replied:
I ended up using that wording all around because we don't actually _perform_ filtering, just identify cells to filter. On one hand I agree about consistency, but on the other hand we aren't actually doing the same thing here! Either way I've accepted suggestions and will fiddle with other wording to be more clear.

allyhawkins commented on templates/qc_report/cite_qc.rmd:
````diff
@@ -35,19 +35,114 @@ knitr::kable(adt_information, align = 'r') |>
+
+```{r}
+
+filtered_cell_count <- sum(filtered_sce$adt_scpca_filter == "Remove")
````
```suggestion
filtered_cell_count <- sum(filtered_sce$adt_scpca_filter == "Keep")
```
I think I would capture the keep cells, see my comment below.

allyhawkins commented on templates/qc_report/cite_qc.rmd:
```diff
@@ -35,19 +35,114 @@ knitr::kable(adt_information, align = 'r') |>
+basic_statistics <- tibble::tibble(
+    "Method used to identify cells targeted for filtering" = format(processed_meta$adt_scpca_filter_method),
+    "Number of cells targeted for filtering"  = format(filtered_cell_count),
+    "Percent of cells targeted for filtering" =
```
```suggestion
    "Percent of cells that pass filtering thresholds" =
```

allyhawkins commented on templates/qc_report/cite_qc.rmd:
````diff
@@ -35,19 +35,114 @@ knitr::kable(adt_information, align = 'r') |>
+```
+
+
+## Expression of highly variable ADTs
````
Before you get to this I think we need a plot showing the filtering that is suggested. We could include another plot like the keep/remove colored plot in RNA but color cells by RNA filter, ADT filter, all filter, or keep.

allyhawkins commented on templates/qc_report/cite_qc.rmd:
```diff
@@ -35,19 +35,114 @@ knitr::kable(adt_information, align = 'r') |>
+
+## Expression of highly variable ADTs
+
+The plots below show UMAP embeddings calculated from RNA expression where each cell is colored by the expression level of the labeled 
```
Alternatively (or additionally?) we could show a faceted ridge plot with the top variable ADT tags on the X axis and the Y axis is the normalized expression.

allyhawkins reviewed (commented):
Thanks for making these changes! I left a few comments on where I think we could clarify things and then also how we could add in some more plots that I think would be helpful either in addition to or instead of the UMAPs.

sjspielman commented:
This should be ready for a new look! I've made the following changes:
- Include a faceted plot of all the ADT/RNA combinations of filtering, again showing the miQC plot
  - I had some trouble here with `glue::glue()`! Note the printed statement that will end up being deleted but is here now for debugging... why doesn't `glue::glue()` print inside of the `if`? I played around with this a bunch, and found more generally that none of the glues inside of `if` blocks are printing.. Does anyone have ideas?!
- I fixed a bug with adt output - much of the `has_adt` logic assumed `has_processed` is true, which is not a safe assumption! I updated in some spots to avoid this trap.
- I added a ridgeplot into the cite report, currently living before the UMAPs. Here, I found that the chunk only wants to print out one plot (another confusing rmarkdown thing 🤷‍♀️), so I ended up just using `patchwork` to ensure they are both printed.   
Here's what the report currently looks like: 
[qc_report.html.zip](https://github.com/AlexsLemonade/scpca-nf/files/11871915/qc_report.html.zip)

jashapiro reviewed (commented):
I left a bunch of little comments, some of which are minor formatting, some of which may address your questions, and some of which are just me being picky. 

I think the biggest one is that we probably don't want to use `getHVGs` for ADTs, as there is not enough data to fit the model well. Do we can just calculate variance from the matrix on the fly. Similarly, if there is no `logcounts` matrix because normalization failed, we can still make one for plotting here, which I think is better than skipping the plots. We just need to add a warning that normalization has not been performed.

sjspielman commented on bin/post_process_sce.R:
```diff
@@ -203,6 +203,12 @@ if (alt_exp %in% altExpNames(processed_sce)) {
     logcounts(altExp(processed_sce, alt_exp)) <- result_matrix
+    
+    # Also, calculate & save most variable ADTs from `adt_sce` 
+    top_adts <- adt_sce |>
+      scran::modelGeneVar() |>
+      scran::getTopHVGs()
+    metadata(processed_sce)$highly_variable_adts <- top_adts
```
> I don't think we want to do modeling here: I don't think there is enough data

:100: makes more sense to me.

sjspielman commented on templates/qc_report/cite_qc.rmd:
```diff
@@ -35,19 +35,210 @@ knitr::kable(adt_information, align = 'r') |>
+      labs(
+        color = "Log-normalized ADT expression"
+      ) +
+      theme(legend.position = "bottom", 
```
> Also, let's maybe remove numbers from the UMAP?

I can do that all around!

sjspielman replied:
done in https://github.com/AlexsLemonade/scpca-nf/pull/343/commits/fac803cda9c5519a70f3cb60449cdd623948ba6f & https://github.com/AlexsLemonade/scpca-nf/pull/343/commits/f51e0466dd4355fc942b2e80272b70c9c5503348

sjspielman commented on templates/qc_report/cite_qc.rmd:
```diff
@@ -35,19 +35,210 @@ knitr::kable(adt_information, align = 'r') |>
+         title = "Combined RNA and ADT filters") +
+    scale_color_viridis_d(direction = -1) +
+    facet_wrap(vars(filter_summary)) +
+    theme(plot.title = element_text(hjust = 0.5),
+          legend.position = "none")
```
Yep, got this all into a `theme_set()` on my end already when I was making plotting changes during your review!

sjspielman commented on templates/qc_report/cite_qc.rmd:
```diff
@@ -35,19 +35,210 @@ knitr::kable(adt_information, align = 'r') |>
+            legend.text = element_text(size = 8)) + 
+      guides(colour = guide_colorbar(title.position = "bottom", title.hjust = 0.5)) 
+    
+    # Combine with patchwork so that both plots are reliably printed
+    adt_ridgeplot / adt_umap
```
No need, can easily separate and that will make for clearer alt text to add in as well.

sjspielman replied:
done in https://github.com/AlexsLemonade/scpca-nf/pull/343/commits/f51e0466dd4355fc942b2e80272b70c9c5503348

sjspielman commented on templates/qc_report/cite_qc.rmd:
```diff
@@ -35,19 +35,210 @@ knitr::kable(adt_information, align = 'r') |>
+    adt_ridgeplot <- ggplot(var_adt_exp_df, 
+           aes(y = ADT, x = adt_expression, fill = ADT)) + 
+      ggridges::geom_density_ridges(alpha = 0.5) + 
+      scale_fill_viridis_d() + 
```
Yes, I also thought of this while you were reviewing and I was making plot changes! I ended up changing to use a 2 color scheme for the faceted keep/remove plot to _match_ the keep/remove colors in the earlier cell filtering plot that is in the main QC report. For this plot, I ended up just using default ggplot fills (since empty density plots look weird to me).

sjspielman commented on templates/qc_report/cite_qc.rmd:
```diff
@@ -35,19 +35,210 @@ knitr::kable(adt_information, align = 'r') |>
+      as.data.frame() |>
+      tibble::rownames_to_column("barcode") |>
+      # combine all ADTs into a single column for easy faceting 
+      tidyr::pivot_longer(cols = starts_with("CD"),
```
> not make an assumption about ADT names?

:+1:

sjspielman replied:
done in https://github.com/AlexsLemonade/scpca-nf/pull/343/commits/f51e0466dd4355fc942b2e80272b70c9c5503348

sjspielman commented on templates/qc_report/cite_qc.rmd:
```diff
@@ -35,19 +35,210 @@ knitr::kable(adt_information, align = 'r') |>
+      theme(legend.position = "none") 
+    
+    
+    # Next, extract umap embeddings as a dataframe
+    umap_df <- reducedDim(processed_sce, "UMAP") |>
+      as.data.frame() |>
+      tibble::rownames_to_column("barcode") |>
+      rename("UMAP1" = "V1",
+             "UMAP2" = "V2") 
+    
+    # combine ADT expression with coldata, umap embeddings, and create data frame to use for plotting
+    coldata_df <- colData(processed_sce) |>
+      as.data.frame() |>
+      tibble::rownames_to_column("barcode") |>
+      # combine with gene expression
+      left_join(var_adt_exp_df, by = "barcode") |>
+      # combine with umap embeddings 
+      left_join(umap_df, by = "barcode")
```
👀

allyhawkins commented on templates/qc_report/cite_qc.rmd:
````diff
@@ -35,19 +35,210 @@ knitr::kable(adt_information, align = 'r') |>
+```{r fig.height=10, fig.width=8, message=FALSE, warning=FALSE, results='asis'}
+if (has_processed) {
+  
+  # only create plot if variable ADTs are present
+  if (!is.null(processed_meta$highly_variable_adts)) {
````
For non-CITE-seq samples, the backup is to log transform the counts if something fails with the default normalization method. On the one hand, I like the idea of mirroring that and including log-transformed counts for CITE-seq. On the other hand, I am wary of providing what may appear like normalized counts to users and them using those without re-doing normalization. I think ultimately I'm in favor of adding it to the object rather than failing normalization, but we definitely need to make sure documentation is up to date.

sjspielman replied:
>  I think ultimately I'm in favor of adding it to the object rather than failing normalization, but we definitely need to make sure documentation is up to date.

👍

sjspielman replied:
Hm, I think we might also want another metadata variable here to match RNA.... `adt_normalization = <median-based/log-normalization>`

allyhawkins replied:
> Hm, I think we might also want another metadata variable here to match RNA.... `adt_normalization = <median-based/log-normalization>`

I agree with this.

sjspielman replied:
done in https://github.com/AlexsLemonade/scpca-nf/pull/343/commits/c566fdd62711087930028ca0e45c24fe01e33182
I directly used `log1p` here, and added the metadata string.

sjspielman commented on templates/qc_report/cite_qc.rmd:
```diff
@@ -35,19 +35,210 @@ knitr::kable(adt_information, align = 'r') |>
+glue::glue("
+  This glue is OUTSIDE the if and it prints just fine. 
+  Where are my glues INSIDE the ifs? :( 
+  ")
```
done in https://github.com/AlexsLemonade/scpca-nf/pull/343/commits/f51e0466dd4355fc942b2e80272b70c9c5503348

sjspielman commented on templates/qc_report/qc_report.rmd:
```diff
@@ -28,9 +28,10 @@ knitr::opts_chunk$set(
 # Set default ggplot theme
-theme_set(theme_bw() + 
+theme_set(theme_bw() +
           theme(plot.margin = margin(rep(20, 4))))
```
done in https://github.com/AlexsLemonade/scpca-nf/pull/343/commits/ab677de1bb8b5e26894d1e3ce49bb0ddb63f5374

sjspielman commented on templates/qc_report/cite_qc.rmd:
```diff
@@ -35,19 +35,210 @@ knitr::kable(adt_information, align = 'r') |>
+                            values_to = "adt_expression")
+    
+    # First, expression ridgeplot ridgeplots
+    adt_ridgeplot <- ggplot(var_adt_exp_df, 
```
done in https://github.com/AlexsLemonade/scpca-nf/pull/343/commits/f51e0466dd4355fc942b2e80272b70c9c5503348

sjspielman commented:
Thanks @jashapiro and @allyhawkins a lot for all your comments! I've responded to everything here, and we're moving right along. I've tested workflow changes as well.

At this point, this PR is getting ~too big. I suggest, unless there is a _major_ problem, let's merge this in soon and if there are more smaller changes needed we can tackle them in a subsequent PR?

jashapiro commented on templates/qc_report/cite_qc.rmd:
```diff
@@ -35,19 +35,206 @@ knitr::kable(adt_information, align = 'r') |>
+  # Calculate variance for each ADT
+  adt_var <- altExp(processed_sce) |>
+    logcounts() |>
+    as.matrix() |> 
```
Not sure this is needed: `apply` functions should work fine on `Matrix` objects too. But I could be wrong,

jashapiro commented on templates/qc_report/cite_qc.rmd:
```diff
@@ -35,19 +35,206 @@ knitr::kable(adt_information, align = 'r') |>
+
+  # grab expression for top ADTs from counts
+  var_adt_exp_df <- logcounts(altExp(processed_sce))[top_adts,] |>
+    as.matrix() |>
+    t() |>
```
Likely a bit more efficient to swap the order here and let the `Matrix` package work its magic.
```suggestion
    t() |>
    as.matrix() |>
```

jashapiro commented on templates/qc_report/cite_qc.rmd:
```diff
@@ -35,19 +35,206 @@ knitr::kable(adt_information, align = 'r') |>
+    apply(1, var, na.rm=TRUE) 
+  
+  # Get the top 4
+  top_adts <- adt_var[ order(adt_var, decreasing=TRUE)[1:top_n] ] |>
+    names()
```
Annoying that `order` doesn't preserve the names!

sjspielman replied:
rude indeed!!! will re-look at this..

sjspielman replied:
Seems ok to me actually, see example - 

```
> vec <- c("A" = 1.5, "B" = 2.5, "C" = 0.5, D = "10")
> vec[ order(vec, decreasing=TRUE)[1:2] ]
    B     D 
"2.5"  "10" 
> vec[ order(vec, decreasing=TRUE)[1:2] ] |> names()
[1] "B" "D"
```

jashapiro replied:
No, it is fine! I was thinking of doing things in a different way with `vec |> order() |> head() |> names()` to save some nesting, but you can't do that because `order()` drops the names.

sjspielman replied:
gotcha :)

jashapiro commented on templates/qc_report/cite_qc.rmd:
```diff
@@ -35,19 +35,206 @@ knitr::kable(adt_information, align = 'r') |>
+    labs(
+      color = "Log-normalized ADT expression"
+    ) +
+    # remove axis numbers and background grid
+    scale_x_continuous(labels = NULL, breaks = NULL) + 
+    scale_y_continuous(labels = NULL, breaks = NULL) + 
```
Did you try adding in `coord_fixed()`? I'm thinking we want to add that to all UMAP plots to make them plot with the same aspect ratio.

sjspielman replied:
thanks for the reminder here!

jashapiro commented on templates/qc_report/qc_report.rmd:
```diff
@@ -28,10 +28,13 @@ knitr::opts_chunk$set(
-theme_set(theme_bw() + 
-          theme(plot.margin = margin(rep(20, 4))))
+theme_set(
+  theme_bw() +theme(plot.margin = margin(rep(20, 4)), 
+                    strip.background = element_rect(fill = 'transparent'))
```
Just formatting
```suggestion
  theme_bw() +
  theme(plot.margin = margin(rep(20, 4)), 
        strip.background = element_rect(fill = 'transparent'))
```

jashapiro reviewed (approved):
This looks good to me, with some very minor suggestions, except for the `coord_fixed()` one, which I think we should do as long as it doesn't look awful.

allyhawkins commented on templates/qc_report/cite_qc.rmd:
````diff
@@ -35,19 +35,206 @@ knitr::kable(adt_information, align = 'r') |>
+```
+
+
+The plot below displays UMAP embeddings calculated from RNA expression, where each cell is colored by the expression level of the given ADT. 
````
Should this be below the next chunk? I think you want it right before the UMAP is actually plotted?

sjspielman replied:
yes good catch, thanks!

allyhawkins reviewed (approved):
I had one minor comment about the placement of the UMAP statement. Other than that I'm going to approve, since I'm trusting you're fixing all of Josh's suggestions. I don't need to see this again.

sjspielman commented:
Here's the "final" version of the report for posterity - 
[qc_report.html.zip](https://github.com/AlexsLemonade/scpca-nf/files/11898907/qc_report.html.zip)

Main notable change is that using `coord_fixed()` means the plot display area needs to be more explicitly sized, so facet labels aren't cut off. The cutting off could still happen, but you'd need a gene symbol that's _longer_ than an Ensembl ID, which is not going to be very likely, and the benefit from `coord_fixed()` seemed greater.
